services/core/DataMover/datamover/agent.py

```diff
@@ -0,0 +1,233 @@
+                                              message_string=message[0]))
+                raise
+
+            if topic_replace_list:
+                if topic in self._topic_replace_map.keys():
+                    topic = self._topic_replace_map[topic]
+                else:
+                    self._topic_replace_map[topic] = topic
+                    temptopics = {}
+                    for x in topic_replace_list:
+                        if x['from'] in topic:
+                            new_topic = temptopics.get(topic, topic)
+                            temptopics[topic] = new_topic.replace(
+                                x['from'], x['to'])
+
+                    for k, v in temptopics.items():
+                        self._topic_replace_map[k] = v
+                    topic = self._topic_replace_map[topic]
+
+            payload = {'headers': headers, 'message': data}
+
+            self._event_queue.put({'source': "forwarded",
+                                   'topic': topic,
+                                   'readings': [(timestamp_string, payload)]})
+
+        def publish_to_historian(self, to_publish_list):
+            _log.debug("publish_to_historian number of items: {}"
+                       .format(len(to_publish_list)))
+            current_time = self.timestamp()
+            last_time = self._last_timeout
+            _log.debug('Last timeout: {} current time: {}'.format(last_time,
+                                                                  current_time))
+            if self._last_timeout:
+                # if we failed we need to wait 60 seconds before we go on.
+                if self.timestamp() < self._last_timeout + 60:
+                    _log.debug('Not allowing send < 60 seconds from failure')
+                    return
+            if not self._target_platform:
+                self.historian_setup()
+            if not self._target_platform:
+                _log.debug('Could not connect to target')
+                return
+
+            to_send = []
+            for x in to_publish_list:
+                to_send.append({'topic': x['topic'],
+                                'headers': x['value']['headers'],
+                                'message': x['value']['message']})
+
+            try:
+                self._target_platform.vip.rpc.call(destination_historian_identity,
+                                                   'insert', to_send).get(timeout=10)
+                self.report_all_handled()
+            except gevent.Timeout:
+                self._last_timeout = self.timestamp()
+                self._target_platform.core.stop()
+                self._target_platform = None
+                self.vip.health.set_status(
+                    STATUS_BAD, "Timout occured")
+
+        def historian_setup(self):
+            _log.debug(
+                "Setting up to forward to {}".format(destination_vip))
+            event = gevent.event.Event()
```

Comment:
Use build connection to create connection directly to the target platforms agent.